diff --git a/cmake/external_libs/nccl.cmake b/cmake/external_libs/nccl.cmake
index d2bc1bca39..9fc5ff3a6e 100644
--- a/cmake/external_libs/nccl.cmake
+++ b/cmake/external_libs/nccl.cmake
@@ -12,8 +12,8 @@ mindspore_add_pkg(nccl
         LIBS nccl
         URL ${REQ_URL}
         MD5 ${MD5}
-        BUILD_OPTION src.build NVCC_GENCODE="-gencode=arch=compute_70,code=sm_70"
+        BUILD_OPTION src.build # NVCC_GENCODE="-gencode=arch=compute_70,code=sm_70"
         INSTALL_INCS build/include/*
         INSTALL_LIBS build/lib/*)
 include_directories(${nccl_INC})
-add_library(mindspore::nccl ALIAS nccl::nccl)
\ No newline at end of file
+add_library(mindspore::nccl ALIAS nccl::nccl)
