diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0b61c1987f..252e053ae1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,6 +53,7 @@ include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/third_party/securec/include)
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/third_party/flatbuffers/include)
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/third_party/flatbuffers/include/flatbuffers)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/third_party/kungfu/include)
 
 include(${CMAKE_SOURCE_DIR}/cmake/dependency_utils.cmake)
 find_package(Python3 COMPONENTS Interpreter Development)
diff --git a/mindspore/ccsrc/CMakeLists.txt b/mindspore/ccsrc/CMakeLists.txt
index 76a50095e9..dd2ca92ac5 100644
--- a/mindspore/ccsrc/CMakeLists.txt
+++ b/mindspore/ccsrc/CMakeLists.txt
@@ -4,7 +4,7 @@ include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 include_directories(${CMAKE_BINARY_DIR})
 include_directories(${CMAKE_SOURCE_DIR}/mindspore/core/mindrt/include)
 include_directories(${CMAKE_SOURCE_DIR}/mindspore/core/mindrt/src)
-
+link_directories(${CMAKE_SOURCE_DIR}/third_party/kungfu/lib)
 
 set(SERVER_FLATBUFFER_OUTPUT "${CMAKE_BINARY_DIR}/schema")
 set(FBS_FILES
@@ -429,3 +429,12 @@ if(MODE_ASCEND_ALL)
 endif()
 
 add_subdirectory(cxx_api)
+
+if ($ENV{ENABLE_KUNGFU})
+    target_link_libraries(mindspore kungfu)
+    if (ENABLE_GPU)
+        target_link_libraries(mindspore kungfu_nccl)
+        target_link_libraries(mindspore mindspore::nccl)
+    endif()
+endif()
+
